repos/keg-cli/src/utils/services/publishService.js

```diff
@@ -3,14 +3,33 @@ const { Logger } = require('KegLog')
 const { spawnCmd } = require('KegProc')
 const { ask } = require('@keg-hub/ask-it')
 const { copySync, emptyDirSync } = require('KegFileSys/fileSys')
-const { get } = require('@keg-hub/jsutils')
+const { get, exists } = require('@keg-hub/jsutils')
 const { getHubRepos } = require('../hub/getHubRepos')
 const { versionService } = require('./versionService')
 const { generalError } = require('../error/generalError')
 const { runRepoScript } = require('../hub/runRepoScript')
 const { getPublishContext } = require('../publish/getPublishContext')
 const { getPublishContextOrder } = require('../publish/getPublishContextOrder')
 const { getVersionUpdate, getValidSemver } = require('KegUtils/version')
+
+/**
+ * Validates the package command and runs it if exists
+ * @param {Object} repo 
+ * @param {Boolean} runTask - to run the task or not
+ * @param {string} scriptName - name of the script cmd to run
+ * @param {Function} errorCB - called when the script being executed throws an error
+ * 
+ * @returns {Void}
+ */
+const validatePublishTask = async (repo, runTask, scriptName, errorCB) => {
+  logFormal(repo, `${runTask ? 'Running' : 'Skipping'} yarn ${scriptName}...`)
+  if (runTask) {
+    exists(get(repo, `package.scripts.${scriptName}`, undefined))
```

Comment:
Pretty sure you don't need the `undefined` in the `get` call but I could be wrong.

Reply:
you're correct. i just verified it.
will update before merging